src/main/java/ua/vedroid/dao3/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,206 @@
+                PreparedStatement statement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getLong("id"));
+            }
+            if (!car.getDrivers().isEmpty()) {
+                addDrivers(connection, car);
+            }
+            return car;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Insert " + car + " to database is failed", ex);
+        }
+    }
+
+    @Override
+    public Optional<Car> getById(Long id) {
+        String query = "SELECT * FROM cars "
+                + "JOIN manufacturers m on m.id = cars.m_id "
+                + "LEFT JOIN cars_drivers cd on cars.id = cd.car_id "
+                + "LEFT JOIN drivers d on d.id = cd.driver_id "
+                + "WHERE cars.id = ?"
+                + " AND cars.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getCarFromResultSet(connection, resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Get car with id="
+                    + id + " is failed", ex);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT * FROM cars JOIN manufacturers m on m.id = cars.m_id "
+                + "WHERE cars.is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(connection, resultSet));
+            }
+            return cars;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Get all cars is failed", ex);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET m_id = ?, model = ?"
+                + " WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            removeDrivers(connection, car.getId());
+            statement.setLong(1, car.getManufacturer().getId());
```

Comment:
Are you sure you want to set id manually and mess with database autoincrement system?..

Reply:
this won't have any effect, let's remove this. You are updating BY id, so id remains the same.